Modules/Microsoft365DSC/DSCResources/MSFT_SCProtectionAlert/MSFT_SCProtectionAlert.psm1

```diff
@@ -0,0 +1,676 @@
+        [System.Boolean]
+        $NotifyUserOnFilterMatch,
+
+        [Parameter()]
+        [System.DateTime]
+        $NotifyUserSuppressionExpiryDate,
+
+        [Parameter()]
+        [System.Int32]
+        $NotifyUserThrottleThreshold,
+
+        [Parameter()]
+        [System.Int32]
+        $NotifyUserThrottleWindow,
+
+        [Parameter()]
+        [System.String[]]
+        $NotifyUser = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $Operation = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $PrivacyManagementScopedSensitiveInformationTypes = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $PrivacyManagementScopedSensitiveInformationTypesForCounting = @(),
+
+        [Parameter()]
+        [System.UInt64]
+        $PrivacyManagementScopedSensitiveInformationTypesThreshold,
+
+        [Parameter()]
+        [System.String]
+        $Scenario,
+
+        [Parameter()]
+        [ValidateSet('Low', 'Medium', 'High')]
+        [System.String]
+        $Severity = 'Low',
+
+        [Parameter()]
+        [System.String]
+        $StreamType,
+
+        [Parameter()]
+        [System.String]
+        $TagFilter,
+
+        [Parameter()]
+        [ValidateSet('Activity', 'Malware', 'Phish', 'Malicious','MaliciousUrlClick')]
+        [System.String]
+        $ThreatType,
+
+        [Parameter()]
+        [ValidateRange(3,2147483647)]
+        [System.Int32]
+        $Threshold,
+
+        [Parameter()]
+        [ValidateRange(60,2147483647)]
+        [System.Int32]
+        $TimeWindow,
+
+        [Parameter()]
+        [System.String[]]
+        $UserTags = @(),
+
+        [Parameter()]
+        [System.Int32]
+        $VolumeThreshold
+    )
+
+
+    Write-Verbose -Message "Setting configuration of SCProtectionAlert for $Name"
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'SecurityComplianceCenter' `
+        -InboundParameters $PSBoundParameters
+
+    $CurrentAlert = Get-TargetResource @PSBoundParameters
+
+    if ($CurrentAlert.IsSystemRule) {
+        Write-Verbose "Amending or creating a 'SystemRule' is currently not supported. "
+    }
+    elseif (('Present' -eq $Ensure) -and ('Absent' -eq $CurrentAlert.Ensure))
+    {
+        $CreationParams = $PSBoundParameters
+        $CreationParams.Remove('Credential')
+        $CreationParams.Remove('Ensure')
+        $CreationParams.Remove('IsSystemRule')
+        $CreationParams.Remove('Scenario')
+        $CreationParams.Remove('StreamType')
```

Comment:
What are all these parameters for if we don't support them in the create or update scenarios?

Reply:
TheIsSystemRule to find those system rules that cannot be adjusted via set or new.. 
(write message). The other parameters could be removed yes - they should be exportable tough to distinguish for example that is a system rule vs a custom rule. besides the "isSystemRule", the other parameters I can remove from the get and set if you think it makes no sense to have them?

Reply:
As a rule of thumb, parameters that are included in the resource cannot be read only. While we understand that from a reporting perspective it may be good to include them, it gives the false impression that they support CRUD operations. We should remove those from the resource if they cannot be used in the New/Set/Update scenarios. That also includes the IsSystemRule property if we cannot use it in the New scenario

Reply:
okey got it. have remove those parameters now